Framework/PythonInterface/plugins/algorithms/CreatePoleFigureTableWorkspace.py

```diff
@@ -0,0 +1,231 @@
+        self.no_chi2_needed = chi2_thresh < 1e-6
+        # if the x0_threshold is set to allow all peaks, no x0 column is needed
+        self.no_x0_needed = x0_thresh < 1e-6
+        if peak_ws:
+
+            def check_col(col_name):
+                try:
+                    peak_ws.column(col_name)
+                except RuntimeError:
+                    issues["PeakParameterWorkspace"] = f"PeakParameterWorkspace must have column: '{col_name}'"
+
+            # if peak parameter ws is given, expect these columns
+            check_col("I")
+
+            if not self.no_x0_needed:
+                check_col("X0")
+
+            if not self.no_chi2_needed:
+                check_col("chi2")
+
+        # check that if a reflection is given it can be parsed to V3D
+        if not self.getProperty("Reflection").isDefault:
+            try:
+                _parse_hkl(self.getProperty("Reflection").value)
+            except:
+                issues["Reflection"] = "Problem parsing hkl string, ensure H,K,L are separated by commas"
+
+            # if a hkl is given, scattering power will be calculated, check that both a sample and CrystalStructure have
+            # been defined
+            try:
+                ws.sample().getShape().volume()  # python api hasShape doesn't work as expected
+                # this will check if there is a valid sample
+                if not ws.sample().hasCrystalStructure():
+                    issues["InputWorkspace"] = "If reflection is specified: InputWorkspace sample must have a CrystalStructure"
+            except RuntimeError:
+                issues["InputWorkspace"] = "If reflection is specified: InputWorkspace must have a sample"
+        return issues
+
+    def PyExec(self):
+        # get inputs
+        ws = self.getProperty("InputWorkspace").value
+        peak_param_ws = self.getProperty("PeakParameterWorkspace").value
+        hkl = self.getProperty("Reflection").value
+        hkl_is_set = not self.getProperty("Reflection").isDefault
+        chi_thresh = self.getProperty("Chi2Threshold").value
+        x0_thresh = self.getProperty("PeakPositionThreshold").value
+
+        # generate a detector table to get detector positions
+        det_table = self.exec_child_alg(
+            "CreateDetectorTable", InputWorkspace=ws, IncludeDetectorPosition=True, DetectorTableWorkspace="det_table"
+        )
+        det_pos = np.asarray(det_table.column("Position"))
+
+        # get the source pos for ki
+        source_pos = np.asarray(ws.getInstrument().getSource().getPos())
+
+        # get the rotation matrix of the sample
+        rot_mat = ws.run().getGoniometer().getR()
+
+        # matrix to put detectors in sample frame
+        to_pole_view = np.linalg.inv(rot_mat)
+
+        # get Qs
+        qds = det_pos / np.linalg.norm(det_pos, axis=1)[:, None]
+        # normalise source pos then take away from sample pos for ki
+        qi = np.zeros(3) - source_pos / np.linalg.norm(source_pos)
```

Comment:
In principle the sample position doesn't have to be at (0,0,0)
```suggestion
        sample_pos = np.asarray(ws.getInstrument().getSample().getPos())
        qi = sample_pos - source_pos / np.linalg.norm(source_pos)
```
I would actually keep most vector arithmetic as V3Ds and only convert to numpy arrays when needed, but that's just personal preference!

Reply:
Actually.... [V3D.norm()](https://docs.mantidproject.org/nightly/api/python/mantid/kernel/V3D.html#mantid.kernel.V3D.norm) is already a method that exists
```python
sample_pos = ws.getInstrument().getSample().getPos()
qi = sample_pos - source_pos / source_pos.norm()
```
should give the equivalent

Reply:
I'm actually not sure I want to use the sample position by default... For scanning texture measurements, where the sample is moving around a static gauge volume on (0,0,0) using the position would introduce some (very small) errors? My proposal is a `UseSamplePosition` flag which is set to false by default and when false assumes the beam is centred on (0,0,0) and only if true uses sample position?

Reply:
Late comment: The sample position (where the sample is) and scattering center (the position that the beam is built around) should be the same place. This is one of the things that a good experimental setup should do.

Reply:
Thanks @peterfpeterson for looking at this and merging, sorry for the delay! Normally I'd agree, but in this case we're moving the sample position on the workspace when we move the sample shape required for attenuation corrections for spatially resolved texture analysis. So really what we're after here is the sample position in the IDF before we've moved it! I wonder if the two things should actually be decoupled (sample position and scattering center)? 
Anyway, I think in practice hard-coding (0,0,0) would have been OK, but adding an extra input is playing it safe (workflow/reduction very much a work-in-progress). I think it would only matter if e.g. only the edge of the sample was illuminated.